lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1446,3 +1458,59 @@ class EditableTextGeometry {
       ..transform = cssTransform;
   }
 }
+
+/// Controls the capitalization of the text.
+///
+/// This corresponds to Flutter's [TextCapitalization].
+///
+/// Uses `text-transform` css property.
+/// See: https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform
+enum TextCapitalization {
+  /// Uppercase for the first letter of each word.
+  words,
+
+  /// Currently not implemented on Flutter Web. Uppercase for the first letter
+  /// of each sentence.
+  sentences,
+
+  /// Uppercase for each letter.
+  characters,
+
+  /// Lowercase for each letter.
+  none,
+}
+
+/// Helper class for text capitalization.
+///
+/// Uses `text-transform` css property.
+/// See: https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform
+class TextCapitalizationUtil {
```

Comment:
I think I can also use extension methods here to add more to the TextCapitalization enum.